### PR TITLE
Add initial REST API specs

### PR DESCRIPTION
**Pull request by eloaverona**
This PR adds initial openapi spec doc for the Grid REST API. 
It defines endpoint to submit batches, get batch status, and get schema from the reporting database. 
To visualize the document [Swagger](http://editor.swagger.io/) is a good tool.

### CODE REVIEW

File: daemon/openapi.yaml [hunk 1]
```diff
@@ -0,0 +1,393 @@
...
+# ------------------------------------------------------------------------------
+
+swagger: '2.0'
+
+info:
+  version: "0.1.0"
+  title: Grid REST API
+  description:
+    _An API providing HTTP/JSON interface to Hyperledger Grid simplifying client
+    interaction with the blockchain by exposing endpoints that use common
```
Comment (agunde406) [inline daemon/openapi.yaml]: Since Grid could be used without a blockchain, we should remove the mention here.

File: daemon/openapi.yaml [hunk 2]
```diff
@@ -0,0 +1,393 @@
...
+    HTTP/JSON standards._
+
+paths:
+  /batches:
+    post:
+      tags:
+      - sawtooth_validator
+      summary: Sends a BatchList to the validator
```
Comment (agunde406) [inline daemon/openapi.yaml]: Mention Sawtooth Validator instead of just validator

File: daemon/openapi.yaml [hunk 3]
```diff
@@ -0,0 +1,393 @@
...
+  wait:
+    name: wait
+    in: query
+    type: integer
+    description: A time in seconds to wait for commit
+
+definitions:
+  Link:
+    type: string
+    example: https://api.sawtooth.com/state?head=65cd3a3ce088b265b626f704b7f3db97b6f12e848dccb35d7806f3d0324c71b709ed360d602b8b658b94695374717e3bdb4b76f77886953777d5d008558247dd
```
Comment (agunde406) [inline daemon/openapi.yaml]: replace https://api.sawtooth.com with https://api.grid.com

File: daemon/openapi.yaml [hunk 4]
```diff
@@ -0,0 +1,393 @@
...
+    properties:
+      name:
+        type: string
+        example: "Lightbulb"
+      description:
+        type: string
+        example: "Example Lightbulb schema"
+      owener:
+        type: string
+        example: "philips"
```
Comment (agunde406) [inline daemon/openapi.yaml]: if we want to match the rfc this could be philips001

File: daemon/openapi.yaml (hunk 1 above)
Comment (eloaverona) [reply daemon/openapi.yaml]: Addressed

File: daemon/openapi.yaml (hunk 2 above)
Comment (eloaverona) [reply daemon/openapi.yaml]: Addressed

File: daemon/openapi.yaml (hunk 3 above)
Comment (eloaverona) [reply daemon/openapi.yaml]: Addressed

File: daemon/openapi.yaml (hunk 4 above)
Comment (eloaverona) [reply daemon/openapi.yaml]: Addressed

**Comment by vaporos** [comment]
Please add a note in the commit messages which describe the documents these are derived from, for reference purposes.

**Pull request merged**